src/storage-client/src/controller.rs

```diff
@@ -1469,13 +1474,20 @@ where
                 )]));
 
                 if frontier.is_empty() {
-                    self.state.pending_host_deprovisions.insert(id);
+                    let status_id =
+                        self.state.introspection_ids[&IntrospectionType::SourceStatusHistory];
+                    self.state.pending_host_deprovisions.insert((id, status_id));
                 }
             }
         }
 
         let to_deprovision = std::mem::take(&mut self.state.pending_host_deprovisions);
-        for id in to_deprovision {
+        for (id, status_id) in to_deprovision {
+            // Record the destruction of the source
+            let status_row = healthcheck::pack_status_row(id, "dropped", None, (self.state.now)());
+            self.append_to_managed_collection(status_id, vec![(status_row, 1)])
```

Comment:
Any particular reason for this ordering? Instead of `deprovision` first?

Reply:
Not any strong reason! It seemed worse to have no `dropped` event for a dropped hosts than duplicate `dropped` events if the drop was interrupted, but I don't feel passionately about it.

Reply:
:ok_hand: